util/collate/general_ci.go

```diff
@@ -31,8 +29,20 @@ func sign(i int) int {
 	return 0
 }
 
+// truncateTrailingSpaces truncates trailing spaces.
+func truncateTrailingSpaces(str string) string {
+	b := []byte(str)
+	length := len(b)
+	for length > 0 && b[length-1] == ' ' {
+		length--
+	}
+	b = b[:length]
+	return string(b)
```

Comment:
I don't think we need `string()` and `[]byte(str)` here, at least you can remove one of them.

Since the returned value of this function is the same(or even less) than `str`, I think it's OK to remove both `string()` and `[]byte(str)` without potential the sub-string memory leak, like what you did in `truncateTailingSpaceOne()`.